src/app/manage-deposits/manage-single-deposit/manage-single-deposit.component.ts

```diff
@@ -10,120 +10,145 @@ import {
+                        swal('', error.status, 'warning');
+                    }
                     SpinnerUtil.hideSpinner();
-                    swal('', 'Success', 'success');
-                }, hideSpinnerAndDisplayError);
-        }, hideSpinnerAndDisplayError);
+                    this.removeUppyPlugin();
+                }
+            }, hideSpinnerAndDisplayError);
     }
 
     getDeposit() {
-        const id = this.route.snapshot.params.ID;
         SpinnerUtil.showSpinner();
+        const id = this.route.snapshot.params.ID;
         this.depositCooperativeService.getDeposit(id).subscribe(res => {
+            SpinnerUtil.hideSpinner();
             this.depositModel = res;
-
-            if (!this.depositModel.deposit.approved) {
-                setTimeout(() => {
-                    this.createUploadArea();
-                    MicroModal.init();
-                }, 300);
-            } else {
+            if (this.depositModel.deposit.approved) {
                 this.generateSignerAndSign();
             }
-
-            SpinnerUtil.hideSpinner();
         }, hideSpinnerAndDisplayError);
     }
 
-    approveButtonClicked() {
-        if (this.amount !== this.amountToConfirm) {
-            swal('', 'The deposit amounts don\'t match. Please check the proper deposit amount and try again!',
-                'error').then(() => {
-                (<any>$('#modal-confirm-deposit')).modal('hide');
-                location.reload();
-            });
-            return;
-        }
-
-        const depositApprovalURL = this.depositCooperativeService.generateDepositApprovalURL(
-            location.origin,
-            this.depositModel.deposit.id,
-            this.amount
-        );
-
-        this.paymentUppy.use(Uppy.XHRUpload, {
-            endpoint: depositApprovalURL,
-            fieldName: 'file',
-            headers: {
-                'Authorization': this.http.authHttpOptions().headers.get('Authorization')
+    showConfirmModal() {
+        this.confirmationModal = this.modalService.show(ManageSingleDepositModalComponent, {
+            initialState: {
+                depositAmount: this.amount
             }
         });
+        const confirmationSub = this.confirmationModal.content.successfulConfirmation
+            .subscribe(() => {
+                const depositApprovalURL = this.depositCooperativeService.generateDepositApprovalURL(
+                    location.origin,
+                    this.depositModel.deposit.id,
+                    this.amount
+                );
+                this.paymentUppy.use(Uppy.XHRUpload, {
+                    endpoint: depositApprovalURL,
+                    fieldName: 'file',
+                    headers: {
+                        'Authorization': this.http.authHttpOptions().headers.get('Authorization')
+                    }
+                });
+                confirmationSub.unsubscribe();
```

Comment:
this method won't be triggered in error scenarios (if they even happen). It could be better that it's triggered in `finalize` block of `successfulConfirmation` subscription. Something like this:

```
const confirmationSub = this.confirmationModal.content.successfulConfirmation.pipe(
        finalize(() => confirmationSub.unsubscribe())
    ).subscribe(...)
```

This will be always triggered on every success or failure, so we are assured that this subscription tears down in every scenario.